pkg/security/podsecuritypolicy/provider.go

```diff
@@ -252,6 +252,15 @@ func (s *simpleProvider) ValidatePodSecurityContext(pod *api.Pod, fldPath *field
 				allErrs = append(allErrs, field.Invalid(
 					field.NewPath("spec", "volumes").Index(i), string(fsType),
 					fmt.Sprintf("%s volumes are not allowed to be used", string(fsType))))
+				continue
+			}
+
+			if fsType == extensions.HostPath {
```

Comment:
Looks like this check will be triggered only when `volumes` not equal to `*`. This is incorrect.